### PR TITLE
Combined .gitignore across merged repos.

### PR DESCRIPTION
**Pull request by sadjamz**
Changes proposed in this PR:
- Previously we had two .gitignores, one from the control-plane and one from charts/consul. This PR makes one .gitignore that is basically the set union of these two.
- '.idea/' is also ignored incase anyone uses GoLand or any other JetBrains IDE.

How I've tested this PR:
Manually tried adding files and didn't see them when typing `git status`.

How I expect reviewers to test this PR:
Code review.

Checklist:
- [ ] Bats tests added
- [ ] CHANGELOG entry added (*HashiCorp engineers only, community PRs should not add a changelog entry*)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,6 @@
 terraform.tfstate*
 terraform.tfvars
 values.dev.yaml
+bin/
+pkg/
+.idea/
```
Comment (ndhanushkodi) [inline .gitignore:9]: Should we include control-plane bin and pkg folders too? I think the reason we ignore them is temporary stuff gets created in there in makefile scripts or when we do builds, so might make sense to ignore them inside the module as well
```suggestion
.idea/
control-plane/bin
control-plane/pkg
```

**Review by ndhanushkodi — APPROVED** [review]
Had an additional suggestion, but looks good to me!

File: .gitignore (hunk 1 above)
Comment (ndhanushkodi) [reply .gitignore:9]: maybe it automatically includes control-plane/bin and control-plane/pkg and we don't have to do this? I already see those not being added by git

File: .gitignore [hunk 2]
```diff
@@ -4,3 +4,6 @@
 terraform.tfstate*
 terraform.tfvars
 values.dev.yaml
+bin/
+pkg/
```
Comment (lkysow) [inline .gitignore:8]: Do we have these dirs?

Comment (t-eckert) [reply .gitignore:8]: Wouldn't they be generated by running `go build` and the like?

File: .gitignore (hunk 1 above)
Comment (lkysow) [reply .gitignore:9]: once cli comes in then it'll probably have those dirs too so probably good to have it at the top-level

**Review by lkysow — APPROVED** [review]

**Pull request merged**